docs/src/main/asciidoc/openshift.adoc

```diff
@@ -0,0 +1,176 @@
+      <artifactId>quarkus-openshift</artifactId>
+    </dependency>
+----
+
+By adding this dependency, we now have the ability to configure the Openshift resource generation and application using the usual `application.properties` approach that Quarkus provides.
+The configuration items that are available can be found in: `io.quarkus.kubernetes.deployment.OpenshiftConfig` class.
+Furthermore, the items provided by `io.quarkus.deployment.ApplicationConfig` affect the Openshift resources.
+
+=== Building
+
+Building is handled by the link:container-image#s2i[container-image-s2i] extension. To trigger a build:
+
+[source, subs=attributes+]
+----
+mvn clean package -Dquarkus.container-image.build=true
+----
+
+The command above will trigger an s2i binary build.
+
+=== Deploying
+
+To trigger a deployment:
+
+[source, subs=attributes+]
+----
+mvn clean package -Dquarkus.kubernetes.deploy=true
+----
+
+The command above will trigger a container image build and will apply the generated Openshift resources, right after.
+The generated resources are using Openshift's `DeploymentConfig` that is configured to automatically trigger a redeployment when a change in the `ImageStream` is noticed.
+In other words, any container image build after the inital deployment will automatically trigger redeployment, without the need to delete, update or re-apply the generated resources.
+
+=== Customizing
+
+All available customization options are available in the link:kubenretes#openshift[Openshift configuration options].
+
+Some examples are provided in the sections below:
+
+==== Labels
+
+To add a label in the generated resources:
+
+[source]
+----
+quarkus.openshift.labels.foo=bar
+----
+
+==== Annotations
+
+To add an annotation in the generated resources:
+
+[source]
+----
+quarkus.openshift.annotations.foo=bar
+----
+
+==== Environment variables
+
+To add an annotation in the generated resources:
+
+[source]
+----
+quarkus.openshift.env-vars.my-env-var.value=foobar
+----
+
+The command above will add `MY_ENV_VAR=foobar` as an environment variable.
+Please note that the key `my-env-var` will be converted to uppercase and dashes will be replaced by underscores resulting in `MY_ENV_VAR`.
+
+You may also noticed that in contrast to labels, and annotations for environment variables you don't just use a key=value approach.
+That is because for environment variables there are additional options rather than just value.
+
+===== Environment variables from Secret
+
+To add all key value pairs of a `Secret` as environment variables:
+
+[source]
+----
+quarkus.openshift.env-vars.my-env-var.secret=my-secret
+----
+
+===== Environment variables from ConfigMap
+
+To add all key value pairs of a `ConfigMap` as environment variables:
+
+[source]
+----
+quarkus.openshift.env-vars.my-env-var.configmap=my-secret
+----
+
+==== Exposing Routes
+
+To expose a `Route` for the Quarkus application:
+
+[source]
+----
+quarkus.openshift.expose=true
+----
```

Comment:
should we add a:

TIP: You do not need to always put these in `application.properties` you can also pass it as a command line argument: `-Dquarkus.openshift.expose=true`

Reply:
Agree! Let me add that.